src/main/java/com/ldtteam/structurize/items/ItemScanTool.java

```diff
@@ -241,12 +233,479 @@ public boolean canAttackBlock(final BlockState state, final Level worldIn, final
+            }
+            String name = "";
+            if (parsed.getContext().getArguments().containsKey(ScanCommand.FILE_NAME))
+            {
+                name = StringArgumentType.getString(cmdContext, ScanCommand.FILE_NAME);
+            }
+
+            stack.getOrCreateTag().put(NBT_COMMAND_POS, NbtUtils.writeBlockPos(command.getBlockPos()));
+            stack.getOrCreateTag().putString(NBT_DIMENSION, command.getLevel().dimension().location().toString());
+
+            final ScanToolData data = new ScanToolData(stack.getOrCreateTag());
+            data.setCurrentSlotData(new ScanToolData.Slot(name, new BoxPreviewData(from, to, anchor)));
+            final ScanToolData.Slot slot = loadSlot(data, stack);
+            Network.getNetwork().sendToPlayer(new ShowScanMessage(slot.getBox()), player);
+
+            player.displayClientMessage(Component.translatable("com.ldtteam.structurize.gui.scantool.copy.ok", name), false);
+            player.playNotifySound(SoundEvents.NOTE_BLOCK_CHIME, SoundSource.PLAYERS, 1.0F, 1.0F);
+        }
+        catch (CommandSyntaxException e)
+        {
+            player.displayClientMessage(Component.translatable("com.ldtteam.structurize.gui.scantool.copy.notscan"), false);
+        }
+    }
+
+    /**
+     * Called on server side when shift-middle-clicking a command block.
+     * @param player the player
+     * @param stack the scan tool
+     * @param command the command block
+     * @param modifiers the modifier keys
+     */
+    private void onCommandBlockPaste(@NotNull final ServerPlayer player,
+                                     @NotNull final ItemStack stack,
+                                     @NotNull final CommandBlockEntity command,
+                                     final int modifiers)
+    {
+        final ScanToolData data = new ScanToolData(stack.getOrCreateTag());
+        saveSlot(data, stack, player);
+        final ScanToolData.Slot slot = data.getCurrentSlotData();
+
+        if (slot.getName().isBlank() || slot.getName().contains(" "))
+        {
+            player.sendSystemMessage(Component.translatable("com.ldtteam.structurize.gui.scantool.paste.badname"));
+            player.playNotifySound(SoundEvents.NOTE_BLOCK_BIT, SoundSource.PLAYERS, 1.0F, 1.0F);
+            return;
+        }
+
+        //noinspection StatementWithEmptyBody
+        if (command.getCommandBlock().getCommand().isBlank())
+        {
+            // it's always ok to set a blank command block
+        }
+        else if (!command.getCommandBlock().getCommand().contains(MOD_ID + " " + ScanCommand.NAME + " "))
+        {
+            player.displayClientMessage(Component.translatable("com.ldtteam.structurize.gui.scantool.paste.badcommand"), false);
+            player.playNotifySound(SoundEvents.NOTE_BLOCK_BIT, SoundSource.PLAYERS, 1.0F, 1.0F);
+            return;
+        }
+        else if ((modifiers & GLFW.GLFW_MOD_CONTROL) == 0)
```

Comment:
wut, screen.isCtrl/shift/altDown() please
(or since this seems to server side then copy out it's logic with comment for future reference)

Reply:
It passes the `modifiers` across from the `event.getModifiers()`, so this seems more correct as is.  This isn't a GUI, there is no screen.

Reply:
I was originally hoping to use `player.isSprinting()` but sadly that only triggers when actually sprinting, not merely when holding the sprint key.